DDG4/edm4hep/EDM4hepFileReader.cpp

```diff
@@ -217,6 +222,23 @@ namespace dd4hep::sim {
           ctx->event().addExtension<EventParameters>(parameters);
         } catch(std::exception &) {}
       }
+#if EDM4HEP_BUILD_VERSION >= EDM4HEP_VERSION(0, 99, 3)
+      // Attach the GeneratorEventParameters if they are available
+      const auto &genEvtParameters = frame.get<edm4hep::GeneratorEventParametersCollection>(edm4hep::labels::GeneratorEventParameters);
+      if (genEvtParameters.isValid()) {
+        if (genEvtParameters.size() == 1) {
+          const auto genParams = genEvtParameters[0];
+          try {
+            auto *ctx = context();
+            ctx->event().addExtension(new edm4hep::MutableGeneratorEventParameters(genParams.clone()));
+          } catch (std::exception &) {}
+        } else {
+          printout(WARNING, "EDM4hepFileReader", "Multiple GeneratorEventParameters found in input file. Ignoring all but one!");
```

Comment:
> Ignoring all but one

But now you are ignoring all of them, or am I missing where the one is not ignored?

Or do you mean that you will always ignore GeneratorEventParameters unless there is just one?